core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -402,6 +422,19 @@ class DAGScheduler(
     }
   }
 
+  /**
+   * Check whether the barrier stage requires more slots (to be able to launch all tasks in the
+   * barrier stage together) than the total number of active slots currently. Fail current check
+   * if trying to submit a barrier stage that requires more slots than current total number. If
+   * the check fails consecutively for three times for a job, then fail current job submission.
```

Comment:
Seems I do not find the code about `"consecutively for three times"`, but only `maxFailureNumTasksCheck ` ?